[DO NOT MERGE]Feature/SPRO-21: Pass order details in create subscription API calls

Dmitry-Pryshchepa opened the pull request:
Make changes requested in code review, add declare strict types directive

Commit:
SPRO-21: Pass order details in create subscription API calls

Make changes requested in code review, add declare strict types directive

ytorbyk commented on Model/Order/DetailsCreator.php:
```diff
@@ -0,0 +1,59 @@
+     */
+    public function __construct(
+        \Swarming\SubscribePro\Platform\Service\OrderDetails $orderDetailsService,
+        \SubscribePro\Service\OrderDetails\Builder $orderDetailsBuilder,
```
Why is the orderDetailsBuilder located in SDK? The SDK should not know anything about Magneto order

ytorbyk commented on Observer/Checkout/SubmitAllAfter.php:
```diff
@@ -97,13 +107,34 @@ public function execute(Observer $observer): void
             $this->checkoutSession->setData(SubscriptionCreator::FAILED_SUBSCRIPTION_COUNT, 0);
         }
+
+        if ($this->isSubscriptionsCreated($result)) {
```
`$result` could be undefined if `$this->subscriptionCreator->createSubscriptions($quote, $order)` method throws exception.

ytorbyk commented on Observer/Checkout/SubmitAllAfter.php:
```diff
@@ -97,13 +107,34 @@ public function execute(Observer $observer): void
+     */
+    private function isSubscriptionsCreated(array $result): bool
+    {
+        return isset($result[SubscriptionCreator::CREATED_SUBSCRIPTION_IDS])
+            ? !empty($result[SubscriptionCreator::CREATED_SUBSCRIPTION_IDS])
+            : false;
```
The next condition does exaclty the same.
```suggestion
        return !empty($result[SubscriptionCreator::CREATED_SUBSCRIPTION_IDS]);
```

ytorbyk commented on Platform/Service/OrderDetails.php:
```diff
@@ -0,0 +1,35 @@
+     * @param int|null $websiteId
+     * @return \SubscribePro\Service\DataInterface
+     */
+    public function createOrderDetails(array $orderDetailsData = [], $websiteId = null)
```
Please, add the result type to the method.

ytorbyk commented on Platform/Service/OrderDetails.php:
```diff
@@ -0,0 +1,35 @@
+     * @throws \SubscribePro\Exception\EntityInvalidDataException
+     * @throws \SubscribePro\Exception\HttpException
+     */
+    public function saveOrderDetails(OrderDetailsInterface $orderDetails, $websiteId = null)
```
Please, add the result type to the method.

ytorbyk reviewed (changes requested):
@Dmitry-Pryshchepa , we should be careful with adding `declare(strict_types=1);` to all existing files. PHP will not cast types automatically with the flag and it can cause errors.
We need to be sure we do not break existing logic or fix it if it's necessary.

Commit:
SPRO-21: Pass order details in create subscription API calls

Remove unnecessarily added strict types directives

Commit:
SPRO-21: Pass order details in create subscription API calls

remove unnecessarily removed use declarations

Commit:
SPRO-21: Pass order details in create subscription API calls

merge in the master branch

Commit:
SPRO-21: Pass order details in create subscription API calls

Make code review requested changes, move OrderDetailsBuilder to Swarming\SubscribePro

Commit:
SPRO-21: Pass order details in create subscription API calls

Change return type for OrderDetails methods, wrap creating order details in try-catch
